src/1.generic-constraints/index.ts

```diff
@@ -4,6 +4,18 @@
 // You can remove the readonly modifier from the mapped type by using -readonly
 
 // ------- start coding here -------
+function mapObjectProperties<
+  S extends object,
+  M extends { [K in keyof M]: keyof S }
+>(source: S, mapping: M): { [K in keyof M]: S[M[K]] } {
```

Comment:
```suggestion
>(source: S, mapping: M): { [K in keyof M]: S[M[K]] } {
```
Kode berikut `{ [K in keyof M]: S[M[K]] }`, silahkan dibuat menggunakan sebuah type sendiri yang nantinya akan dipanggil dalam function atau tidak dibuat secara literal. 
Tips: pada type yang sudah dibuat, silahkan menggunakan `-readonly` untuk menghilangkan readonly

Reply:
Sudah saya commit ulang ya, mohon di review